sdk/provisioning/Azure.Provisioning/src/Primitives/Resource.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Azure.Core;
+using Azure.Provisioning.Expressions;
+
+namespace Azure.Provisioning.Primitives;
+
+/// <summary>
+/// Represents an Azure resource.
+/// </summary>
+/// <param name="resourceName">The friendly namee of the resource.</param>
+/// <param name="resourceType">The type of the resource.</param>
+/// <param name="resourceVersion">The version of the resource.</param>
+/// <param name="context">Optional ProvisioningContext.</param>
+public abstract class Resource(string resourceName, ResourceType resourceType, string? resourceVersion = default, ProvisioningContext? context = default)
+    : NamedProvisioningConstruct(resourceName, context)
+{
```

Comment:
`Resource` is the pinnacle of the `ProvisioningConstruct` line by adding a resourceType and resourceVersion to the mix.  Most of the core types you interact with in these libraries will derive from `Resource`.